### ISSUE TITLE
move simulate method out of SigningStargateClient class to make sure knowing private key is not a mandatory requirement to simulate

### ISSUE THREAD

**Issue by harpreet-mehra-coindcx** [cosmos/stargate#101]
`async simulate(signerAddress, messages, memo) {
        const anyMsgs = messages.map((m) => this.registry.encodeAsAny(m));
        const accountFromSigner = (await this.signer.getAccounts()).find((account) => account.address === signerAddress);
        if (!accountFromSigner) {
            throw new Error("Failed to retrieve account from signer");
        }
        const pubkey = (0, amino_1.encodeSecp256k1Pubkey)(accountFromSigner.pubkey);
        const { sequence } = await this.getSequence(signerAddress);
        const { gasInfo } = await this.forceGetQueryClient().tx.simulate(anyMsgs, memo, pubkey, sequence);
        (0, utils_1.assertDefined)(gasInfo);
        return math_1.Uint53.fromString(gasInfo.gasUsed.toString()).toNumber();
    }`

The above function `simulate` can be invoked by `SigningStargateClient` which is a client created with an RPC connection URL and a key pair. Simulate transactions should not need the use of a key pair but only the use of a signer address should suffice. Note that in the function definition of the `simulate`, only the signer's public address is used which is how it should be. So please remove the requirement that the invoking object should need a private key.

**Comment by oviawork** [comment]
yes simulate function should be available in stargateClient and not SigningStargateClient where private key is not mandatory to initialize the object
